File: testprojects/scala_main/pipelines/LIVY_SCALA/code/src/main/scala/io/prophecy/pipelines/livy_scala/graph/FlattenSchema_1.scala
package io.prophecy.pipelines.livy_scala.graph

import io.prophecy.libs._
import io.prophecy.pipelines.livy_scala.config.ConfigStore._
import io.prophecy.pipelines.livy_scala.config.Context
import io.prophecy.pipelines.livy_scala.udfs.UDFs._
import io.prophecy.pipelines.livy_scala.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object FlattenSchema_1 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.withColumn("arr_str1", explode_outer(col("arr_str1")))
      .withColumn("arr_str2", explode_outer(col("arr_str2")))
      .withColumn("struct_complex-col2",
                  explode_outer(col("struct_complex.col2"))
      )
      .withColumn("struct_complex-col1",
                  explode_outer(col("struct_complex.col1"))
      )
      .select(col("arr_str1"),
              col("arr_str2"),
              col("struct_complex.col2").as("struct_complex-col2"),
              col("struct_complex.col1").as("struct_complex-col1"),
              col("year")
      )

}

File: testprojects/python_main/pipelines/LIVY_PYTHON/code/livy_python/graph/FlattenSchema_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from livy_python.config.ConfigStore import *
from livy_python.udfs.UDFs import *

def FlattenSchema_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn("a2", explode_outer("a2"))\
        .withColumn("a3", explode_outer("a3"))\
        .withColumn("a1", explode_outer("a1"))\
        .withColumn("s1-col2", explode_outer("s1.col2"))\
        .withColumn("s1-col1", explode_outer("s1.col1"))\
        .select(col("a1"), col("a2"), col("a3"), col("s1.col2").alias("s1-col2"), col("s1.col1").alias("s1-col1"), col("year"))
